common/extended_window.hpp

```diff
@@ -176,5 +181,27 @@ namespace ILLIXR {
             */
 #endif
         }
+
+        ~xlib_gl_extended_window() {
+            RAC_ERRNO_MSG("xlib_gl_extended_window at start of destructor");
+
+            [[maybe_unused]] const bool gl_result = static_cast<bool>(glXMakeCurrent(dpy, None, nullptr));
+            assert(gl_result && "glXMakeCurrent should not fail");
+
+            /// Destroyed by XCloseDisplay
+            /// See [documentation](https://tronche.com/gui/x/xlib/display/XCloseDisplay.html)
+            // XDestroyWindow(dpy, win);
+            // Window root = DefaultRootWindow(dpy);
+            // XDestroyWindow(dpy, root);
+            // glXDestroyContext(dpy, glc);
+            // XFreeColormap(dpy, _m_cmap);
```

Comment:
Leave with comment, or remove?

Reply:
Leave a comment listing the destroyed resources. No need to have the actual destructor functions.

Reply:
I am not sure that `XCloseDisplay` actually cleans up [_everything_](https://community.khronos.org/t/properly-destroying-a-window/47476). Even [Khronos' own example](https://www.khronos.org/opengl/wiki/Programming_OpenGL_in_Linux:_GLX_and_Xlib) explicitly cleans up the attached resources before closing the display. So I think we should too. I'm finding conflicting information regarding the order of the frees, but I would imagine freeing in LIFO order should be the way to go: context, window, color map, display.